mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/client/DefaultOperationParameters.java

```diff
@@ -31,10 +28,14 @@
 
   private final String configRef;
   private final Map<String, Object> parameters;
+  private final Map<String, Pair<Class, OperationParameters>> complexParams;
 
-  DefaultOperationParameters(String configRef, Map<String, Object> parameters) {
+  DefaultOperationParameters(String configRef,
+                             Map<String, Object> parameters,
+                             Map<String, Pair<Class, OperationParameters>> complexParams) {
     this.configRef = configRef;
     this.parameters = copyOf(parameters);
+    this.complexParams = copyOf(complexParams);
```

Comment:
why do you need to handle the complex parameters as a separate things instead of a special kind of value?